test/sol6/reserves/kyberFprReserveV2.js

```diff
@@ -0,0 +1,3422 @@
+              let amountWei = new BN(1000);
+              let tx = await tradeAndVerifyDataWithSteps(
+                reserveInst,
+                true,
+                tokenInd,
+                amountWei,
+                user1,
+                false
+              );
+              numberTxsPerStep[tx.numberSteps]++;
+              gasUsedPerStep[tx.numberSteps].iadd(new BN(tx.gasUsed));
+            }
+
+            for(let i = 0; i <= 10; i++) {
+              if (numberTxsPerStep[i] > 0) {
+                console.log(`         Average gas used for buy with ${i} steps: ${Math.floor(1.0 * gasUsedPerStep[i]/numberTxsPerStep[i])}`)
+              }
+            }
+          });
+
+          it("Test few sells with steps", async() => {
+            //init conversion rate
+            await setupConversionRateV2sContractWithSteps();
+            await generalSetupReserveContract(isUsingWallet[t], isUsingWeth[t]);
+
+            if (isUsingWeth[t]) {
+              await reserveInst.setTokenWallet(weth.address, walletForToken, {from: admin});
+              await weth.approve(reserveInst.address, MAX_ALLOWANCE, {from: walletForToken});
+            }
+
+            if (isUsingWallet[t]) {
+              // approve
+              for(let i = 0; i < numTokens; i++) {
+                await reserveInst.setTokenWallet(tokenAdd[i], walletForToken, {from: admin});
+                await reserveInst.approveWithdrawAddress(tokenAdd[i], withdrawAddress, true, {from: admin});
+                await tokens[i].approve(reserveInst.address, MAX_ALLOWANCE, {from: walletForToken});
+              }
+            }
+
+            let tokenInd = 3;
+            let token = tokens[tokenInd];
+            // make a simple buy and sell to init some data first
+            await makeSimpleTrades(tokenInd, isUsingWallet[t], isUsingWeth[t]);
+
+            let gasUsedPerStep = [];
+            let numberTxsPerStep = [];
+            for(let i = 0; i <= 10; i++) {
+              gasUsedPerStep.push(new BN(0));
+              numberTxsPerStep.push(0);
+            }
+            for(let i = 0; i <= 5; i++) {
+              let amountTwei = new BN(6000);
+              await token.transfer(network, amountTwei);
+              await token.approve(reserveInst.address, 0, {from: network});
+              await token.approve(reserveInst.address, amountTwei, {from: network});
+              let tx = await tradeAndVerifyDataWithSteps(
+                reserveInst,
+                false,
+                tokenInd,
+                amountTwei,
+                user1,
+                false
+              );
+              numberTxsPerStep[tx.numberSteps]++;
+              gasUsedPerStep[tx.numberSteps].iadd(new BN(tx.gasUsed));
+            }
+
+            for(let i = 0; i <= 10; i++) {
+              if (numberTxsPerStep[i] > 0) {
+                console.log(`         Average gas used for buy with ${i} steps: ${Math.floor(1.0 * gasUsedPerStep[i]/numberTxsPerStep[i])}`)
```

Comment:
seems here it should be 
used for sell (not buy)

Reply:
ok